src/PeriodEnd/Refunds/SFA.DAS.ProviderPayments.Calc.Refunds/Infrastructure/Repositories/HistoricalPaymentsRepository.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using SFA.DAS.Payments.DCFS.Infrastructure.Data;
+using SFA.DAS.ProviderPayments.Calc.Refunds.Infrastructure.Entities;
+
+namespace SFA.DAS.ProviderPayments.Calc.Refunds.Infrastructure.Repositories
+{
+    public interface IHistoricalPaymentsRepository
+    {
+        List<HistoricalPaymentEntity> GetAllForProvider(long ukprn);
+    }
+
+    public class HistoricalPaymentsRepository : DcfsRepository, IHistoricalPaymentsRepository
+    {
+        public HistoricalPaymentsRepository(string transientConnectionString) 
+            : base(transientConnectionString) { }
+
+        public List<HistoricalPaymentEntity> GetAllForProvider(long ukprn)
```

Comment:
I'm not sure of this because payments has so many different standards, but I think this should be a IEnumerable not list

Reply:
I agree. The usage would be to iterate though them and IEnumerable makes more sense in this context. Also it's what's returned from dapper.

Reply:
worth noting its not dapper that I'm using but the dcfs repo (which yes does use dapper) which returns an array. <- the 90's called, they want their arrays back. meanwhile I've changed it to use IEnumerable<T>.